### PR TITLE
Remove mentions of Py2.7 from docs

### PR DESCRIPTION
**Pull request by Secrus**
# Pull Request Check List

Resolves: #4752 

- [x] Updated **documentation** for changed code.

As decided in #4752, support for managing python 2.7 environments is removed from `poetry`. This PR implements necessary changes in documentation, tests and error handling (as mentioned [here](https://github.com/python-poetry/poetry/issues/4752#issuecomment-1112816106))

### CODE REVIEW

File: src/poetry/utils/env.py [hunk 1]
```diff
@@ -1637,9 +1640,12 @@ def get_pip_command(self, embedded: bool = False) -> list[str]:
         ]
 
     def get_supported_tags(self) -> list[Tag]:
-        output = self.run_python_script(GET_SYS_TAGS)
+        if self.get_version_info() < (3, 0, 0):
+            raise PoetryException("Poetry does not support python 2.7 environments")
```
Comment (abn) [inline src/poetry/utils/env.py:1644]: We can use richer text here, even indicating the plugin. Either via a simple console exception or via `logger.error`.

File: src/poetry/utils/env.py [hunk 2]
```diff
@@ -1637,9 +1640,12 @@ def get_pip_command(self, embedded: bool = False) -> list[str]:
...
+        else:
+            output = self.run_python_script(GET_SYS_TAGS)
```
Comment (abn) [inline src/poetry/utils/env.py:1646]:
```suggestion

        output = self.run_python_script(GET_SYS_TAGS)
```

File: src/poetry/utils/env.py [hunk 3]
```diff
@@ -1637,9 +1640,12 @@ def get_pip_command(self, embedded: bool = False) -> list[str]:
...
 
-        return [Tag(*t) for t in json.loads(output)]
+            return [Tag(*t) for t in json.loads(output)]
```
Comment (abn) [inline src/poetry/utils/env.py:1648]:
```suggestion
        return [Tag(*t) for t in json.loads(output)]
```

File: src/poetry/utils/env.py (hunk 1 above)
Comment (Secrus) [reply src/poetry/utils/env.py:1644]: By richer text you mean more info (including info about plugin)?

**Comment by nikiink** [comment]
As this is a documentation change env.py should be removed. This file is also outdated in pull request.

**Comment by Secrus** [comment]
> As this is a documentation change env.py should be removed. This file is also outdated in pull request.

The scope of this PR changed in the process and the old title is misleading. I am still working on it.

**Pull request closed without merging**